connectors/spark/src/main/scala/io/unitycatalog/spark/auth/GenericCredentialProvider.java

```diff
@@ -92,6 +95,11 @@ void setRenewalLeadTimeMillis(long renewalLeadTimeMillis) {
     this.renewalLeadTimeMillis = renewalLeadTimeMillis;
   }
 
+  // For testing purpose only.
+  void setClock(Clock clock) {
+    this.clock = clock;
```

Comment:
minor: why not take this in the constructor, an overloaded constructor that takes a clock?
then `createTestProvider` can always set it and you dont have to keep calling this method in every test.